Added guide to the git pull command

m-oliv opened the pull request:
- Added guide with info about what the git pull command does, the command syntax and some examples.

HKuz commented on src/pages/git/git-pull/index.md:
```diff
@@ -3,13 +3,31 @@ title: Git Pull
+If you have uncommited changes, the merge part of the *git pull* command will fail and your local branch will be untouched. 
 
+Thus, you should *always commit your changes in a branch before pulling* new commits from a remote repository.
+<!-- The article goes here, in GitHub-flavored Markdown. Feel free to add YouTube videos, images, and CodePen/JSBin embeds  -->
```
Can delete this comment now that you've written the article 👍

HKuz commented on src/pages/git/git-pull/index.md:
```diff
@@ -3,13 +3,31 @@ title: Git Pull
 ## Git Pull
 
-This is a stub. <a href='https://github.com/freecodecamp/guides/tree/master/src/pages/git/git-pull/index.md' target='_blank' rel='nofollow'>Help our community expand it</a>.
+[**git pull**](https://git-scm.com/docs/git-pull) allows you to get new commits from a remote repository. To do this, *git pull* fetches ( *git fetch* ) the new commits and merges [(*git merge*)](https://guide.freecodecamp.org/git/git-merge) these into your local branch.
```
For formatting of the `git ...` commands in this paragraph, let's put them in backticks for syntax highlighting instead of italics/bold. So **git pull** -> `git pull`, ( *git fetch* ) -> (`git fetch`), and (*git merge*) -> (`git merge`). Also, very minor nit, so the first word in the paragraph is capitalized, change "git pull allows you..." to "The `git pull` command allows you..."

HKuz commented on src/pages/git/git-pull/index.md:
````diff
@@ -3,13 +3,31 @@ title: Git Pull
-<a href='https://github.com/freecodecamp/guides/blob/master/README.md' target='_blank' rel='nofollow'>This quick style guide will help ensure your pull request gets accepted</a>.
+This command's syntax is as follows:
+```bash
+git pull [options] [<repository>]
````
We had an older issue with using the angle brackets even in code-fenced examples (they were incorrectly parsed/rendered as HTML, and the article looked weird), so we're using a different all-caps-with-hyphens format to avoid them entirely. This command will look a little goofy, but to be consistent with the other Git articles, change this to `git pull OPTIONS REPOSITORY`. Also, there's a bullet about `refspec` below - should there be a placeholder for it here to avoid confusion?

HKuz commented on src/pages/git/git-pull/index.md:
````diff
@@ -3,13 +3,31 @@ title: Git Pull
+git pull [options] [<repository>]
+```
+in which:
+- **repository** is the URL to your repo. Example:  https://github.com/freeCodeCamp/freeCodeCamp.git
````
For all bullets below, make the word all caps to match the command (it'll look a little odd, but should avoid people having any ambiguity about them). Also, switch the italics for *--quiet* and *--verbose* to use backticks 👍

HKuz commented on src/pages/git/git-pull/index.md:
````diff
@@ -3,13 +3,31 @@ title: Git Pull
+```
+If you want to **pull a specific branch**, you can use:
+```bash
+git fetch <remote_name> <branch_name>
````
Similar comment as above: `<remote_name>` -> `REMOTE-NAME` and `<branch_name>` to `BRANCH-NAME`, and update the bullets below to match.

HKuz commented on src/pages/git/git-pull/index.md:
```diff
@@ -3,13 +3,31 @@ title: Git Pull
+- **remote_name** is the name of your remote repository. For example: *origin*.
+- **branch_name** is the name of your branch. For example: *develop*.
 
+If you have uncommited changes, the merge part of the *git pull* command will fail and your local branch will be untouched. 
```
Final one: *git pull* -> `git pull` 😄

m-oliv commented:
Hi @HKuz!

I've updated the article to use the formatting you suggested. 

Also, you were right, I had forgotten to add the refspec to the command. It is also fixed in the latest commit.

HKuz commented:
LGTM! A+ work, @m-oliv 🐁 🦅 🌳! (And so you know, for inline syntax highlighting in Markdown, you don't have to use the triple backticks - the shortcut is to only use single backticks:  \`git merge\` ) 👍
